models/location_helpers.go

```diff
@@ -274,6 +278,75 @@ func ChangeBackupLocation(q *reform.Querier, locationID string, params ChangeBac
 	return row, nil
 }
 
+func testS3Config(c *S3LocationConfig) error {
+	parsedURL, err := url.Parse(c.Endpoint)
+	if err != nil {
+		return err
+	}
+
+	endpoint := path.Join(parsedURL.Host, parsedURL.Path)
+	secure := true
+	if parsedURL.Scheme == "http" {
+		secure = false
+	}
+	minioClient, err := minio.New(endpoint, c.AccessKey, c.SecretKey, secure)
+	if err != nil {
+		return err
+	}
+
+	exists, err := minioClient.BucketExists(c.BucketName)
+	if err != nil {
+		if er, ok := err.(minio.ErrorResponse); ok {
+			return status.Errorf(codes.InvalidArgument, "%s: %s.", er.Code, er.Message)
+		}
+
+		return err
+	}
+
+	if !exists {
+		return status.Errorf(codes.InvalidArgument, "Bucket doesn't exist")
+	}
+
+	return nil
+}
+
+// TestBackupLocationParams are params for testing location and credentials.
+type TestBackupLocationParams struct {
+	BackupLocationConfig
+}
+
+// TestBackupLocationConfig tests backup location config.
+func TestBackupLocationConfig(params *TestBackupLocationParams) error {
+	configSet, err := params.Validate()
+	if err != nil {
+		return err
+	}
+
+	if !configSet {
+		return status.Error(codes.InvalidArgument, "One config should be set only.")
+	}
+
+	switch {
+	case params.S3Config != nil:
+		if err := checkS3Config(params.S3Config); err != nil {
```

Comment:
or maybe better to not reuse the err variable? Here is my opinion regarding this - in my case the err variable lives in a short scope and it doesn't affect the global state (variables out of `if stmt` scope). If you look at standard library you will see over 100 usages of pattern `if err := ... ; err != nil`, and a few dozens of `if err = ...`.

Reply:
But what is the benefit in your case?

Reply:
https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_use_the_smallest_scope_possible

Reply:
Yes, but there is a `warning` section where shadowing discussed. Here you shadows err variable declared on line 320, it will not cause problems in that particular func, but we have functions that use named returns. 

> will cause two compile errors.
That is a kinda weak argument as for me. Not a big deal to add `:` when it's needed.

Anyway, this topic more about personal preferences, so it's up to you.